tests/test_velocity_verlet_integrator.py

```diff
@@ -0,0 +1,153 @@
+
+    def multiple_steps_U_update(x, v, lamb_sched):
+        # Doesn't use the lamb sched, as multiple_steps_U is always run as a equilibrium simulation
+        context.set_x_t(x)
+        context.set_v_t(v)
+        _, xs, _ = context.multiple_steps_U(0.0, len(lamb_sched), [], 0, 0)
+        v = context.get_v_t()
+        return xs[-1], v
+
+    assert_reversible(x0, v0, multiple_steps_U_update, schedule, atol=atol)
+
+
+def test_reversibility():
+    """Check reversibility of "public" .step and .multiple_steps implementations for a Context using the VelocityVerlet integrator"""
+
+    np.random.seed(2022)
+
+    # define a Python force fxn that calls custom_ops
+    rfe = hif2a_ligand_pair
+    unbound_potentials, sys_params, masses = rfe.prepare_vacuum_edge(rfe.ff.get_ordered_params())
+    coords = rfe.prepare_combined_coords()
+    bound_potentials = [
+        ubp.bind(params).bound_impl(np.float32) for (ubp, params) in zip(unbound_potentials, sys_params)
+    ]
+    box = 100 * np.eye(3)
+
+    dt = 1.5e-3
+
+    # Is not infinitely reversible, will fail after 3000 steps due to accumulation of coords/velos in floating point
+    for n_steps in [1, 10, 100, 500, 1000, 2000]:
+        lamb_sched = np.linspace(0, 1, n_steps)
+        v0 = np.random.randn(*coords.shape)
+        intg, ctxt = setup_velocity_verlet(bound_potentials, coords, box, dt, masses)  # noqa
+        ctxt.set_v_t(v0)
+
+        # check "public" .step and .multiple_steps implementations
+        assert_reversibility_using_step_implementations(ctxt, lamb_sched, atol=1e-10)
+
+
+def test_matches_reference():
+    np.random.seed(2022)
+
+    rfe = hif2a_ligand_pair
+    unbound_potentials, sys_params, masses = rfe.prepare_vacuum_edge(rfe.ff.get_ordered_params())
+    coords = rfe.prepare_combined_coords()
+    box = 100 * np.eye(3)
+
+    dt = 1.5e-3
+
+    lamb = 0.0
+
+    summed_potential = SummedPotential(unbound_potentials, sys_params)
+    summed_potential.bind(np.concatenate([param.reshape(-1) for param in sys_params]))
+    bound_summed = summed_potential.bound_impl(np.float32)
+
+    def force(coords):
+        du_dxs = bound_summed.execute(coords, box, lamb)[0]
+        return -du_dxs
+
+    intg = ReferenceVelocityVerlet(force, masses, dt)
+
+    v0 = np.random.randn(*coords.shape)
+
+    intg_impl, ctxt = setup_velocity_verlet([bound_summed], coords, box, dt, masses)  # noqa
+
+    ctxt.set_v_t(v0)
+
+    n_steps = 10
+
+    # Add one step as the C++ context does N steps + 2 half steps (initialize, finalize)
+    ref_xs, ref_vs = intg.multiple_steps(coords, v0, n_steps=n_steps + 1)
+    np.testing.assert_array_almost_equal(ref_xs[0], coords, decimal=10)
+    np.testing.assert_array_almost_equal(ref_xs[0], ctxt.get_x_t(), decimal=10)
+
+    _, xs, _ = ctxt.multiple_steps(np.ones(n_steps) * lamb, 0, 1)
+    assert xs.shape[0] == n_steps
+    v1 = ctxt.get_v_t()
+    atol = 1e-5
+    np.testing.assert_allclose(ref_xs[-1], xs[-1], atol=atol)
```

Comment:
Why not compare the whole trajectory?
```suggestion
    np.testing.assert_allclose(ref_xs, xs, atol=atol)
```